src/com/esotericsoftware/kryo/Kryo.java

```diff
@@ -389,9 +389,10 @@ public Registration register (Class type) {
 		return register(type, getDefaultSerializer(type));
 	}
 
-	/** Registers the class using the specified ID and the {@link Kryo#getDefaultSerializer(Class) default serializer}. If the ID is
-	 * already in use by the same type, the old entry is overwritten. If the ID is already in use by a different type, a
-	 * {@link KryoException} is thrown. Registering a primitive also affects the corresponding primitive wrapper.
+	/** Registers the class using the specified ID and the {@link Kryo#getDefaultSerializer(Class) default serializer}.
+	 * If the class is already registered this has no effect and the existing registration is returned. Providing an
+	 * ID that is already in use by a different type is unsupported and may result in undefined behavior. Registering a

```

Comment:
I added that line because if you provide a type that has not been registered with an ID that has already been used it won't hit that initial check and will end up implicitly erasing whatever was previously registered with that ID. That is something of an edge case though, so it probably doesn't need to be in the javadocs. I'll remove it here (and in the following ones).